### ISSUE TITLE
SC2027 doesn't account for escaped quotes

### ISSUE THREAD

**Issue by ChemGuy88** [koalaman/shellcheck#2940]
#### For bugs
- Rule Id (if any, e.g. SC1000):  SC2027
- My shellcheck version (`shellcheck --version` or "online"): "online" and "ShellCheck v0.37.0" for VSCode
- [x] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2027)
- [x] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [X] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [X] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/bash
fromPath="path/to/here"
echo "    \`fromPath\`: \""$fromPath"\"."
```

#### Here's what shellcheck currently says:

```
$ shellcheck myscript
 
[Line 3:](javascript:setPosition(3, 28))
echo "    \`fromPath\`: \""$fromPath"\"."
                           ^-- [SC2027](https://www.shellcheck.net/wiki/SC2027) (warning): The surrounding quotes actually unquote this. Remove or escape them.

$
```

#### Here's what I wanted or expected to see:

I expected my syntax to not raise a warning, since I'm doing exactly what it's telling me to do: to escape the quotes. I do not want to disable the warning because it's useful in other situations. I searched the issues on GitHub and the only other work-around i can think of is using unicode quotes, as seen [here](https://github.com/koalaman/shellcheck/issues/689)

Thanks and I appreciate any advice you can give.

**Comment by ChemGuy88** [comment]
Sorry I realized my mistake right after posting this.
